staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -1333,8 +1380,113 @@ func TestNeedsCleanup(t *testing.T) {
 
 }
 
-func TestNeedsUpdate(t *testing.T) {
+// This tests a service update while a slow node sync is happening. If we have multiple
+// services to process from a node sync: each service will experience a sync delta.
+// If a new Node is added and a service is synced while this happens: we want to
+// make sure that the slow node sync never removes the Node from LB set because it
+// has stale data.
+func TestSlowNodeSync(t *testing.T) {
+	stopCh, updateCallCh := make(chan struct{}), make(chan fakecloud.UpdateBalancerCall)
+	defer close(stopCh)
+	defer close(updateCallCh)
+
+	duration := time.Millisecond
```

Comment:
this is very small and use to flake on CI with load , did you run the test with `stress` to verify it does not flake? see instructions in https://gist.github.com/liggitt/6a3a2217fa5f846b52519acfc0ffece0#running-unit-tests-to-reproduce-flakes

Reply:
I've removed the sleet completely and used a channel instead to coordinate the syncs between both controllers.